consensus/src/dag/dag_driver.rs

```diff
@@ -149,11 +153,15 @@ impl RpcHandler for DagDriver {
         {
             let dag_reader = self.dag.read();
             if dag_reader.exists(node.metadata()) {
-                return Ok(CertifiedAck::new(node.metadata().epoch()));
+                return Ok(CertifiedAck::new(epoch));
             }
         }
 
-        self.add_node(node)?;
+        let node_metadata = node.metadata().clone();
+        self.add_node(node).and_then(|_| {
+            self.order_rule.process_new_node(&node_metadata);
```

Comment:
It is not enough to call order_rule here. We should call it every time we enter a node to the dag. For example, certified node response in dag_fetcher. Otherwise, we might lose liveness.